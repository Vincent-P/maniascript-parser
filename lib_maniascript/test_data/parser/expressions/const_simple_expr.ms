#Const C_MyValue -2
#Const C_MyValue -2-1
