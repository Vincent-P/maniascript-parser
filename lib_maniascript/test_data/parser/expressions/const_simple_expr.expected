NODE_ROOT 0..41 {
  NODE_CONST 0..19 {
    TOKEN_CONST("#Const") 0..6
    TOKEN_WHITESPACE(" ") 6..7
    NODE_IDENTIFIER 7..16 {
      TOKEN_IDENT("C_MyValue") 7..16
    }
    TOKEN_WHITESPACE(" ") 16..17
    NODE_UNARY_OP 17..19 {
      TOKEN_MINUS("-") 17..18
      NODE_LITERAL 18..19 {
        TOKEN_INTEGER("2") 18..19
      }
    }
  }
  TOKEN_WHITESPACE("\n") 19..20
  NODE_CONST 20..41 {
    TOKEN_CONST("#Const") 20..26
    TOKEN_WHITESPACE(" ") 26..27
    NODE_IDENTIFIER 27..36 {
      TOKEN_IDENT("C_MyValue") 27..36
    }
    TOKEN_WHITESPACE(" ") 36..37
    NODE_BINARY_OP 37..41 {
      NODE_UNARY_OP 37..39 {
        TOKEN_MINUS("-") 37..38
        NODE_LITERAL 38..39 {
          TOKEN_INTEGER("2") 38..39
        }
      }
      TOKEN_MINUS("-") 39..40
      NODE_LITERAL 40..41 {
        TOKEN_INTEGER("1") 40..41
      }
    }
  }
}